src/cpp/r/RUtil.cpp

```diff
@@ -39,6 +39,20 @@ namespace rstudio {
 namespace r {
 namespace util {
 
+namespace {
+int* s_pErrorNumber;
+} // end anonymous namespace
+
+int errorNumber()
+{
+   return *s_pErrorNumber;
```

Comment:
This will dereference uninitialized memory if `setErrorNumber` is not called, which isn't likely but looks possible if an error occurs while trying to hook it up. How about init'ing the pointer to `nullptr` and returning a safe error number (0?) when the pointer is not initialized?